### PR TITLE
Speed up hmnet inference

### PR DESCRIPTION
**Pull request by chatc**
Add parameters in hmnet model, i.e. `model = HMNetModel(max_gen_length=100, beam_width=2)` will greatly decrease the inference time cost.

### CODE REVIEW

File: model/dialogue/hmnet_model.py [hunk 1]
```diff
@@ -24,10 +24,10 @@ class HMNetModel(SummModel):
     is_extractive = False
     is_neural = True
 
-    def __init__(self):
+    def __init__(self, **kwargs):
```
Comment (niansong1996) [inline model/dialogue/hmnet_model.py:27]: Okay, this leaves almost no information for the users/other team members to know what to input here, hence it's not a good interface.

`**kwargs` is nothing more than a dictionary, when you pass in the args, items that have no direct match in other argument keys will be put into `**kwargs`

Instead, suppose you want to reveal one option (let's call it `some_arg`) and hide the others with `**kwargs`: 
```
def __init__(self, some_arg: some_type = some_default_value, **kwargs):
    kwargs["some_arg"] = some_arg
    # proceed to next lines
```

File: model/dialogue/hmnet_model.py [hunk 2]
```diff
@@ -82,6 +82,14 @@ def _parse_args(self):
             if val is not None:
                 opt[key] = val
 
+        # combine kwargs into opt dictionary (we allow lower case)
+        for key, val in kwargs.items():
+            if key.upper() not in opt.keys():
+                print('WARNING: {} is not a valid key in HMNet.')
```
Comment (niansong1996) [inline model/dialogue/hmnet_model.py:88]: You should put what are the valid keys

**Review by niansong1996 — COMMENTED** [review]
Made some comments, please take a look

File: model/dialogue/hmnet_model.py (hunk 1 above)
Comment (chatc) [reply model/dialogue/hmnet_model.py]: Thanks for the great idea! I've modified the code

File: model/dialogue/hmnet_model.py (hunk 2 above)
Comment (chatc) [reply model/dialogue/hmnet_model.py]: Done!

**Comment by niansong1996** [comment]
@chatc Can you add some comments on what's the typical setting and what speed should people expect.

**Pull request merged**